dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/event/JdbcEventStore.java

```diff
@@ -623,6 +623,10 @@
     @Override
     public List<Map<String, String>> getEventsGrid( EventSearchParams params, List<OrganisationUnit> organisationUnits )
     {
+        User user = currentUserService.getCurrentUser();
+
+        setAccessiblePrograms( user, params );
```

Comment:
I think ideally EventSearchParams should be immutable, but for now this is okay.

Reply:
agree. however, challenge is there are some params/defaults that we need to set so that we avoid some joins or filtering.